extensions/interactions/DragAndDropSortInput/directives/oppia-interactive-drag-and-drop-sort-input.directive.ts

```diff
@@ -93,9 +107,31 @@ angular.module('oppia').directive('oppiaInteractiveDragAndDropSortInput', [
               ctrl.dataMaxDepth = 1;
             }
 
-            // Make list of dicts from the list of choices.
-            for (var i = 0; i < ctrl.choices.length; i++) {
-              ctrl.list.push({title: ctrl.choices[i], items: []});
+            let savedSolution = (
+              $attrs.savedSolution !== undefined ?
+              JSON.parse($attrs.savedSolution) : []
```

Comment:
Why do we need to use the JSON fro saving and retrieving the saved solution? Can we get around this usage somehow?

Reply:
So the thing is, since we are populating the `saved-solution` prop directly via the HTML, there's no way to pass in actual data types since it will all be converted into a string. That is why I've jsonified the data and then parsed the string while extracting it in the directive/component.

Reply:
Can we use something like this https://stackoverflow.com/a/50277474?

Reply:
I'd found the exact same thread while working on this issue!
But, 1, that is for Angular 2+ and most of the interaction files are still in ajs. And 2, we would still have to jsonify the object since we are directly setting the attribute in the HTML as follows:
https://github.com/oppia/oppia/blob/be8759be0ee97a7df4ad1fb48ad344c405aa45ea/core/templates/services/exploration-html-formatter.service.ts#L80-L83

Reply:
Can we add a TODO and issue to refactor this after we complete the Angular migration? In the place that you linked it can be replaced too right?

Reply:
Done! ([Issue filed](https://github.com/oppia/oppia/issues/12292))